### PR TITLE
Add IP address parameter to vhost path

### PR DESCRIPTION
**Pull request by Abdulrazak-Alkl**
Give the ability to HTTP server to verify the client IP address via passing the IP to vhost path.
It will be useful for who has authorization backend vi IP address.

### CODE REVIEW

File: src/rabbit_auth_backend_http.erl [hunk 1]
```diff
@@ -54,9 +54,10 @@ user_login_authorization(Username) ->
         Else                          -> Else
     end.
 
-check_vhost_access(#auth_user{username = Username}, VHost, _Sock) ->
+check_vhost_access(#auth_user{username = Username}, VHost, Sock) ->
     bool_req(vhost_path, [{username, Username},
-                          {vhost,    VHost}]).
+                          {vhost,    VHost},
+			  {ip, inet_parse:ntoa(extract_address(Sock))}]).
```
Comment (hairyhum) [inline src/rabbit_auth_backend_http.erl]: `extract_address/1` can return `undefined`, which will result an error when trying to build the query string.

**Comment by michaelklishin** [comment]
This needs some code formatting (e.g. spaces) and probably can be backported to `stable`.

**Comment by hairyhum** [comment]
@Abdulrazak-Alkl can you please rebase this to `stable` and fix formatting?

**Comment by hairyhum** [comment]
@Abdulrazak-Alkl I have rebased your commits to `stable` in #33

**Pull request closed without merging**